opal/corelib/range.rb

```diff
@@ -126,18 +146,55 @@ def size
   end
 
   def step(n = 1)
-    raise NotImplementedError
+    return enum_for(:step, n) unless block_given?
+    raise ArgumentError.new("step can't be negative") if n < 0
+    raise ArgumentError.new("step can't be 0") unless n > 0
+    if @begin.is_a?(Numeric) and @end.is_a?(Numeric)
```

Comment:
Could you use `Numeric#===` here? `Object#is_a?` can be overwritten, the logic flow must be always defined by `Class#===`